tests/unit/s2n_signature_algorithms_test.c

```diff
@@ -316,47 +315,165 @@ int main(int argc, char **argv)
+            /*
+             * For TLS1.2, always choose either RSA-SHA1 or ECDSA depending on the auth method.
+             */
+            {
+                conn->actual_protocol_version = S2N_TLS12;
+
+                /* For the server signature, the auth method must match the cipher suite. */
+                {
+                    /* Choose RSA for an RSA cipher suite. */
+                    {
+                        conn->secure.cipher_suite = RSA_CIPHER_SUITE;
+                        EXPECT_SUCCESS(s2n_choose_default_sig_scheme(conn, &result, S2N_SERVER));
+                        EXPECT_EQUAL(result.iana_value, s2n_rsa_pkcs1_sha1.iana_value);
+                    }
 
-        s2n_connection_free(conn);
-        s2n_config_free(config);
+                    /* Choose ECDSA for a ECDSA cipher suite. */
+                    {
+                        conn->secure.cipher_suite = ECDSA_CIPHER_SUITE;
+                        EXPECT_SUCCESS(s2n_choose_default_sig_scheme(conn, &result, S2N_SERVER));
+                        EXPECT_EQUAL(result.iana_value, s2n_ecdsa_sha1.iana_value);
+                    }
+
+                    /* Ignore the type of the client certificate. */
+                    {
+                        conn->secure.cipher_suite = ECDSA_CIPHER_SUITE;
+                        conn->handshake_params.client_cert_pkey_type = S2N_PKEY_TYPE_RSA;
+                        EXPECT_SUCCESS(s2n_choose_default_sig_scheme(conn, &result, S2N_SERVER));
+                        EXPECT_EQUAL(result.iana_value, s2n_ecdsa_sha1.iana_value);
+                    }
+
+                    /* When in doubt, choose RSA. */
+                    {
+                        conn->secure.cipher_suite = TLS13_CIPHER_SUITE;
+                        EXPECT_SUCCESS(s2n_choose_default_sig_scheme(conn, &result, S2N_SERVER));
+                        EXPECT_EQUAL(result.iana_value, s2n_rsa_pkcs1_sha1.iana_value);
+                    }
+                }
+
+                /* For the client signature, the auth type must match the type of the client certificate */
+                {
+                    /* Choose RSA for an RSA certificate */
+                    {
+                        conn->handshake_params.client_cert_pkey_type = S2N_PKEY_TYPE_RSA;
+                        EXPECT_SUCCESS(s2n_choose_default_sig_scheme(conn, &result, S2N_CLIENT));
+                        EXPECT_EQUAL(result.iana_value, s2n_rsa_pkcs1_sha1.iana_value);
+                    }
+
+                    /* Choose ECDSA for a ECDSA certificate */
+                    {
+                        conn->handshake_params.client_cert_pkey_type = S2N_PKEY_TYPE_ECDSA;
+                        EXPECT_SUCCESS(s2n_choose_default_sig_scheme(conn, &result, S2N_CLIENT));
+                        EXPECT_EQUAL(result.iana_value, s2n_ecdsa_sha1.iana_value);
+                    }
+
+                    /* Ignore the auth type of the cipher suite */
```

Comment:
This comment isn't correct, right?

Reply:
No, it's correct. You don't consider the cipher suite auth type when choosing a client sig scheme. We base it off the client certificate type instead. Technically we only base the server sig scheme off the cipher suite auth type because the cipher suite auth type is based off of the server certificate type :)